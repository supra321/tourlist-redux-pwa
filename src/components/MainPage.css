@font-face {
font-display: swap;
font-family: 'Futura PT Medium';
font-style: normal;
font-weight: normal;
src: local('Futura PT Medium'), url('../containers/futura-pt-medium-589e45b956de4.woff') format('woff');
}
h1{
	font-family: 'Futura PT Medium';
	font-weight: 200;
	color: rgba(0,0,0,0.65);
}