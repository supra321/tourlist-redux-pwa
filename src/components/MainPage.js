import React,{Component} from 'react';
import {Placelists} from './Placelists';
import Placeset from './Placeset';
import './MainPage.css'
import Header from './Header';
import Scroll from './Scroll';
import Searchbar from './Searchbar';
import Searchbar1 from './Searchbar1';
import ErrorBoundary from '../containers/ErrorBoundary';
class MainPage extends Component{
	componentDidMount() {
    	this.props.onFetchPlacelist();
  	}
  	filteredplaces=Placelists=>{
  		return Placelists.filter(placelist=>{
			return placelist.destination.toLowerCase().includes(this.props.searchBarDestination.toLowerCase())&&placelist.state.toLowerCase().includes(this.props.searchBarState.toLowerCase());
		})
  	}
	render(){
		const {onSearchChangeDestination,onSearchChangeState,placelists,isPending}=this.props;
		console.log(placelists);
		return (
			<div className='tc'>
				<Header />
				<Searchbar1 onSearchChange={onSearchChangeState} />
				<Searchbar onSearchChange={onSearchChangeDestination} />
				<br/>
				<br/>
				<Scroll>
					{
						isPending?<h1 className='f-3 tc'>Loading Please Wait!!!</h1>:
						<ErrorBoundary>
							<Placeset Placelists={this.filteredplaces(Placelists)} />
						</ErrorBoundary>
					}
				</Scroll>
			</div>
		);
	}
}
export default MainPage;