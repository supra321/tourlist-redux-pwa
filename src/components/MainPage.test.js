import React from 'react';
import {shallow} from 'enzyme';
import MainPage from './MainPage';

import Srinagar_Image from '../media/images/SRINAGAR.jpg';
import Srinagar_PDF from '../media/pdf/Srinagar.pdf';

let wrapper;

beforeEach(()=>{
	const mockProps={
		onFetchPlacelist: jest.fn(),
		placelists: [],
		searchBarDestination: '',
		searchBarState: '',
		isPending: false
	}
	wrapper=shallow(<MainPage {...mockProps}/>)
})

it('Expect to render MainPage Component',()=>{
	expect(wrapper).toMatchSnapshot();
})

it('Filters Places Correctly',()=>{
	const mockProps2={
		onFetchPlacelist: jest.fn(),
		placelists: [],
		searchBarDestination: 'Srinagar',
		searchBarState: 'Jammu and Kashmir',
		isPending: false
	}
	const wrapper2=shallow(<MainPage {...mockProps2}/>)
	expect(wrapper2.instance()
		.filteredplaces([
		{
			id: 1,
		    destination: 'Srinagar',
		    state: 'Jammu and Kashmir',
		    pakage: '18 Tours & Activity',
		    pdf: Srinagar_PDF,
		    image: Srinagar_Image
		}
	]))
	.toEqual([
		{
			id: 1,
		    destination: 'Srinagar',
		    state: 'Jammu and Kashmir',
		    pakage: '18 Tours & Activity',
		    pdf: Srinagar_PDF,
		    image: Srinagar_Image
		}
	]);
	expect(wrapper.instance().filteredplaces([])).toEqual([]);
})

it('Filters Places Correctly 2',()=>{
	const mockProps3={
		onFetchPlacelist: jest.fn(),
		placelists: [],
		searchBarDestination: 'Sravasti',
		searchBarState: 'Uttar Pradesh',
		isPending: false
	}
	const wrapper3=shallow(<MainPage {...mockProps3}/>)
	expect(wrapper3.instance()
		.filteredplaces([
		{
			id: 1,
		    destination: 'Srinagar',
		    state: 'Jammu and Kashmir',
		    pakage: '18 Tours & Activity',
		    pdf: Srinagar_PDF,
		    image: Srinagar_Image
		}
	]))
	.toEqual([]);
})
