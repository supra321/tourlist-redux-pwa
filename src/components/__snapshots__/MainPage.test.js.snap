// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render MainPage Component 1`] = `ShallowWrapper {}`;
